src/eterna/ui/Toolbar.ts

```diff
@@ -145,676 +386,3015 @@ export default class Toolbar extends ContainerObject {
+            this.checkDragElement(e.data.global.x, e.data.global.y);
+        }
+    }
 
-                this.regs.add(this._annotationManager.viewAnnotationDataUpdated.connect(() => {
-                    this.annotationPanel.updatePanel();
-                }));
-            }
+    public getMirrorButton(bt: GameButton): GameButton {
+        if (bt.name) {
+            let b = this._bottomButtons.get(bt.name);
+            if (b) return b;
+            b = this.topButtons.get(bt.name);
+            if (b) return b;
         }
+        return new GameButton();
+    }
 
-        if (this._type === ToolbarType.PUZZLEMAKER) {
-            this.submitButton.tooltip('Publish your puzzle!');
-            this.lowerToolbarLayout.addHSpacer(SPACE_WIDE);
-            this.addObject(this.submitButton, this.lowerToolbarLayout);
+    public getMirrorTopButton(bt: GameButton): GameButton | undefined {
+        if (bt.name) {
+            const b = this.topButtons.get(bt.name);
+            if (b) return b;
         }
+        return undefined;
+    }
 
-        this.rightArrow = this.makeArrowButton('right');
-        this.addObject(this.rightArrow, this.scrollContainerContainer);
+    public getMirrorBottomButton(bt: GameButton): GameButton | undefined {
+        if (bt.name) {
+            const b = this._bottomButtons.get(bt.name);
+            if (b) return b;
+        }
+        return undefined;
+    }
 
-        let interval: NodeJS.Timeout;
-        const endScroll = () => {
-            if (interval) clearInterval(interval);
-        };
-        const scrollRight = () => {
-            this.scrollContainer.setScroll(this.scrollContainer.scrollX + 10, this.scrollContainer.scrollY);
-            this.updateArrowVisibility();
-        };
-        const scrollLeft = () => {
-            this.scrollContainer.setScroll(this.scrollContainer.scrollX - 10, this.scrollContainer.scrollY);
-            this.updateArrowVisibility();
-        };
-        this.regs.add(this.rightArrow.pointerDown.connect(() => {
-            endScroll();
-            interval = setInterval(scrollRight, 100);
-        }));
-        this.regs.add(this.rightArrow.pointerUp.connect(() => endScroll()));
-        this.regs.add(this.leftArrow.pointerDown.connect(() => {
-            endScroll();
-            interval = setInterval(scrollLeft, 100);
-        }));
-        this.regs.add(this.leftArrow.pointerUp.connect(() => endScroll()));
-        this.regs.add(this.rightArrow.pointerTap.connect(() => {
-            endScroll();
-            scrollRight();
-        }));
-        this.regs.add(this.leftArrow.pointerTap.connect(() => {
-            endScroll();
-            scrollLeft();
-        }));
+    private collapse() {
+        this._isExpanded = false;
+        this.collapseButton.container.visible = false;
+        this.expandButton.container.visible = true;
+        this._tabsHContainer.visible = false;
+
+        this.removeNamedObjects('test');
```

Comment:
Probably want to give it a better name - and probably use a constant like is done for the location task in the autohide handler

Reply:
fixed